src/Rules.Framework/ConditionTypeExtractor.cs

```diff
@@ -40,23 +46,31 @@ public IEnumerable<TConditionType> GetConditionTypes(IEnumerable<Rule<TContentTy
             {
                 var rootCondition = rule.RootCondition;
 
-                VisitConditionNode(rootCondition, conditionTypes);
+                try
+                {
```

Comment:
Isn't it enough to verify if `rootCondition` is null?

With this approach, you are expecting and making the `NullReferenceException` and `NotSupportedException` a normal behavior, and you shouldn't for various reasons:
* Having "normal" logic working based on exceptions has a big performance penalty. Exceptions are a detour from the normal program execution that cost immensely on the execution.
* `NullReferenceException` can happen due to a null `rootCondition` or due to a bug on the implementation. There's no way to be sure and this logic will "hide" situations where this exception might be thrown due to a bug in the implementation.
* `NotSupportedException` would also be thrown in the scenario of a not supported condition node, which also might expose a bug on the implementation.

Reply:
I think I misunderstood your last comment. I've updated the method to validate if rootcondition is null or not.

Reply:
Thank you @Daniel-C-Dias! 🙂